### PR TITLE
add `Catalog::{contains,find,remove}`

### PR DESCRIPTION
**Pull request by rushmorem**

### CODE REVIEW

**Comment by rushmorem** [comment]
Don't merge yet. I've just thought of one more method I want to add.

**Comment by rushmorem** [comment]
This is now ready for review.

/cc @bluejekyll

File: server/src/authority/catalog.rs [hunk 1]
```diff
@@ -420,18 +430,16 @@ impl Catalog {
...
-            let name = name.base_name();
-            if !name.is_root() {
-                return self.find_auth_recurse(&name);
-            }
-        }
-
-        None
+    /// Recursively searches the catalog for a matching authority
+    pub fn find(&self, name: &LowerName) -> Option<&RwLock<Authority>> {
+        self.authorities.get(name)
```
Comment (bluejekyll) [inline server/src/authority/catalog.rs]: Nice cleanup... you’re touching some of my earliest Rust code ;)

File: server/src/authority/catalog.rs [hunk 2]
```diff
@@ -298,6 +303,11 @@ impl Catalog {
         }
     }
 
+    /// Checks whether the `Catalog` contains DNS records for `name`
```
Comment (bluejekyll) [inline server/src/authority/catalog.rs]: Can you ad a comment about the difference between 'contains' and 'find'.

Something to the effect of this must match exactly, where as find will search the name backwards from most specific until least to find a match zone...

Comment (rushmorem) [reply server/src/authority/catalog.rs]: Done :)

Comment (bluejekyll) [reply server/src/authority/catalog.rs]: perfect, thanks!

**Comment by bluejekyll** [comment]
There's a pesky test-failure that seems to be unrelated to your change... I'm trying to confirm it.

**Pull request merged**